lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering.py

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_virtualnetworkpeering
+version_added: "2.8"
+short_description: Manage Azure Virtual Network Peering.
+description:
+    - Create, update and delete Azure Virtual Network Peering.
+
+options:
+    resource_group:
+        description:
+            - Name of a resource group where the vnet exists.
+        required: true
+    name:
+        description:
+            - Name of the virtual network peering.
+        required: true
+    virtual_network:
+        description:
+            - Virtual network to be peered.
+            - It can be name of virtual network.
+            - It can be virtual network resource id.
+        required: true
+    remote_virtual_network:
+        description:
+            - Remote virtual network to be peered.
+            - It can be name of remote virtual network in same resource group.
+            - It can be remote virtual network resource id.
+            - It can be a dict which contains C(name) and C(resource_group) of remote virtual network.
+            - Required when creating
+    allow_virtual_network_access:
+        description:
+            - Allows VMs in the remote VNet to access all VMs in the local VNet.
+        type: bool
+        default: false
+    allow_forwarded_traffic:
+        description:
+            - Allows forwarded traffic from the VMs in the remote VNet.
+        type: bool
+        default: false
+    use_remote_gateways:
+        description:
+            - Name of the remote virtual network to be peered.
```

Comment:
Incorrect description; "If remote gateways can be used on this virtual network."